### ISSUE TITLE
SQL unique constraint exception parse error (langid = 23)

### ISSUE THREAD

**Issue by letarak** [DataObjects-NET/dataobjects-net#286]
If database/user language is 23 (British English) UniqueConstraintException (2601) cannot be parsed
ErrorMessageParser get isEnglish = false and templates not prepared correctly
On parse try get this exception, because Indexes and parseResult is empty 

```
The given key '1' was not present in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at Xtensive.Sqls.SqlServer.Driver.Driver.TryProvideErrorContext(Int32 errorCode, String errorMessage, SqlExceptionInfo info) in /_/Orm/Xtensive.Orm.SqlServer/Sql.Drivers.SqlServer/Driver.cs:line 70
   at Xtensive.Sqls.SqlServer.Driver.Driver.GetExceptionInfo(Exception exception) in /_/Orm/Xtensive.Orm.SqlServer/Sql.Drivers.SqlServer/Driver.cs:line 35
   at Xtensive.Orm.Providers.StorageExceptionBuilder.BuildException(Exception origin, String queryText) in /_/Orm/Xtensive.Orm/Orm/Providers/StorageExceptionBuilder.cs:line 57
   at Xtensive.Orm.Providers.StorageDriver.ExecuteCommand[TResult](Session session, DbCommand command, CommandBehavior commandBehavior, Func`3 action) in /_/Orm/Xtensive.Orm/Orm/Providers/StorageDriver.Operations.cs:line 448
   at Xtensive.Orm.Providers.Command.ExecuteReader() in /_/Orm/Xtensive.Orm/Orm/Providers/CommandProcessing/Command.cs:line 69
   at Xtensive.Orm.Providers.BatchingCommandProcessor.ExecuteBatch(Int32 numberOfTasks, QueryRequest lastRequest, CommandProcessorContext context) in /_/Orm/Xtensive.Orm/Orm/Providers/CommandProcessing/BatchingCommandProcessor.cs:line 160
   at Xtensive.Orm.Providers.BatchingCommandProcessor.ExecuteTasksWithReader(QueryRequest request, CommandProcessorContext context) in /_/Orm/Xtensive.Orm/Orm/Providers/CommandProcessing/BatchingCommandProcessor.cs:line 86
   at Xtensive.Orm.Providers.SqlSessionHandler.Xtensive.Orm.Providers.IProviderExecutor.ExecuteTupleReader(QueryRequest request, ParameterContext parameterContext) in /_/Orm/Xtensive.Orm/Orm/Providers/SqlSessionHandler.IProviderExecutor.cs:line 27
   at Xtensive.Orm.Rse.RecordSetReader.Prepare(Boolean executeAsync) in /_/Orm/Xtensive.Orm/Orm/Rse/RecordSetReader.cs:line 143
   at Xtensive.Orm.Rse.RecordSetReader.Create(EnumerationContext context, ExecutableProvider provider) in /_/Orm/Xtensive.Orm/Orm/Rse/RecordSetReader.cs:line 228
   at Xtensive.Orm.Linq.TranslatedQuery.ExecuteSequence[T](Session session, ParameterContext parameterContext) in /_/Orm/Xtensive.Orm/Orm/Linq/TranslatedQuery.cs:line 69
   at Xtensive.Orm.Linq.QueryProvider.Execute[TResult](Expression expression, Func`4 runQuery) in /_/Orm/Xtensive.Orm/Orm/Linq/QueryProvider.cs:line 111
   at Xtensive.Orm.Linq.Queryable`1.GetEnumerator() in /_/Orm/Xtensive.Orm/Orm/Linq/Queryable.cs:line 40
   at Xtensive.Orm.Linq.Queryable`1.System.Collections.IEnumerable.GetEnumerator() in /_/Orm/Xtensive.Orm/Orm/Linq/Queryable.cs:line 44
```

**Comment by letarak** [comment]
Version 7.0.3
To reproduce need spend time to prepare code and run mssql with specific locale (I am not very hardworking guy)
Maybe point to the problem piece of code would be enough? :)

```
DriverFactory.CreateMessageParser

46: command.CommandText = LangIdQuery;
47: isEnglish = command.ExecuteScalar().ToString()=="0"; // it is only true for 'us_english'
```
```
ErrorMessageParser

236: templates = isEnglish
237:        ? messageTemplates.ToDictionary(item => item.Key, item => PrepareEnglishTemplate(item.Value)) 
238:        : messageTemplates.ToDictionary(item => item.Key, item => PrepareNonEnglishTemplate(item.Value));
```

PrepareEnglishTemplate and PrepareNonEnglishTemplate told us 
```
// English messages use '%ls' and '%.*ls' as format placeholders.
// They are not indexed. Index is taken from argument position.
      
// Non-english messages use placeholders in form '%N!' where N is argument index.
// We parse only single-digit numbers for simplicity.
```
it is true
query return single language - 1033
```
SELECT language_id FROM SYS.MESSAGES where text like '%*ls%' group by language_id
```

but it is also true, that messages with 1033 message language used for 2 server languages
```
exec sp_helplanguage

lang_id 0   us_english
lang_id 23  British
```

**Comment by alex-kulakov** [comment]
Thank you for the details, I appreciate it. Now it is on me to find non-US MS SQL Instance and check.

**Comment by alex-kulakov** [comment]
This (#314) should be enough. I've checked the rest of languages for how they define parameters and current English Template and Non-English template of ErrorMessageParser are still valid.

**Comment by alex-kulakov** [comment]
I've merged it to ```6.0``` branch then it will show up in ```7.0``` and finally in ```master```.
